include/accessor.h

```diff
@@ -477,6 +495,34 @@ namespace detail {
 		}
 	}
 
+	template <typename DT, int Dims, cl::sycl::access::mode M, cl::sycl::target T, int Index>
+	class multidim_subscriptor {
+		using AccessorT = celerity::accessor<DT, Dims, M, T>;
+
+	  public:
+		multidim_subscriptor(const AccessorT& acc, const size_t d0) : acc(acc), d0(d0), d1(4) {}
+
+		multidim_subscriptor(const AccessorT& acc, const size_t d0, const size_t d1) : acc(acc), d0(d0), d1(d1) {}
+
+		template <int D = Dims>
+		std::enable_if_t<Index != 0 && D == 3, multidim_subscriptor<DT, Dims, M, T, 0>> operator[](const size_t dx) const {
```

Comment:
Instead of those `enable_if` overloads, we should have three specializations of the class. This will not duplicate much code, but will avoid keeping `d1` around which is unused in the 2D case (and initialized to `4` for some reason?). Also, without that change, the constructors would require an `enable_if` as well for completeness.

Reply:
I agree with this suggestion, however, we actually only need two specializations: One for the final 3D proxy (storing both `d0` and `d1`) as well as one for the final 2D / intermediate 3D proxy (storing only `d0`).